src/wp-includes/post.php

```diff
@@ -5690,45 +5690,63 @@ function get_page( $page, $output = OBJECT, $filter = 'raw' ) {
 function get_page_by_path( $page_path, $output = OBJECT, $post_type = 'page' ) {
 	global $wpdb;
 
-	$last_changed = wp_cache_get_last_changed( 'posts' );
-
-	$hash      = md5( $page_path . serialize( $post_type ) );
-	$cache_key = "get_page_by_path:$hash:$last_changed";
-	$cached    = wp_cache_get( $cache_key, 'posts' );
-	if ( false !== $cached ) {
-		// Special case: '0' is a bad `$page_path`.
-		if ( '0' === $cached || 0 === $cached ) {
-			return;
-		} else {
-			return get_post( $cached, $output );
+	$use_wp_query = ! has_filter( 'posts_pre_query' ) && ! has_action( 'pre_get_posts' );
```

Comment:
```suggestion
	/*
	 * Only use WP_Query if not called from the 'posts_pre_query' or 'pre_get_posts' hook,
	 * to avoid an infinite loop. Otherwise, perform a direct SQL query.
	 */
	$use_wp_query = ! has_filter( 'posts_pre_query' ) && ! has_action( 'pre_get_posts' );
```
Let's add a comment here to explain the logic.

Reply:
Should this be `doing_filter` / `doing_action`? 

`has_action()` is an alias for `has_filter()`  so we can simply use `has_action()` if it doesn't need to be a `doing_*` function.

Reply:
Didn't even know that those functions exists. Let me look into this.